core/services/job/models.go

```diff
@@ -859,9 +860,18 @@ func (w *WorkflowSpec) Validate() error {
 		return fmt.Errorf("incorrect length for id %s: expected %d, got %d", w.WorkflowID, workflowIDLen, len(w.WorkflowID))
 	}
 
+	// special case empty owner and name for anonymous workflows
+	if w.WorkflowOwner == "" && w.WorkflowName == "" {
+		return nil
+	}
+
```

Comment:
If we include this, then there should be test cases, which I can't find.

Reply:
good catch, this was a bad push that was half baked btw our call and slack update. removed.